replace README with rewritten one (includes JSON support, request headers)

davewhiteland opened the pull request:
Rewritten README was lagging behind changes so this is the rewrite as a single
commit. (Includes changes that had also been reflected in commits to README
for `Scraped::JSON` and request header passing example).

This does *not* include a separate `example.rb` file for tests, and it drops the
illustration which had non-exemplar example code snippets in it.

Commit:
rewrite README (includes JSON support and request headers)

Rewritten README was lagging behind changes so this is the rewrite as a single
commit. (Includes changes that had also been refelected in commits to README
for Scraped::JSON and request header passing example).

This doesn't include a separate example.rb file for tests, and it drops the
illustration which had non-exemplar example code snippets in it.

Commit:
Merge remote-tracking branch 'origin' into 45-refactor-readme-single-edit

to pass Rubocop

tmtmtmtm commented on README.md:
```diff
@@ -1,10 +1,12 @@
-Write declarative scrapers in Ruby
+Write declarative scrapers in Ruby.
+
+If you need to write a webscraper (especially one that hits a page which lists a load of other pages, and jumps into each of _those_ pages to pull out the same data) the `scraped` gem will help you write it quickly and clearly.
```
Does that 'especially' go too far in that direction? I find `scraped` really useful even when it's only scraping a single page.

tmtmtmtm commented on README.md:
```diff
@@ -20,14 +22,29 @@ Or install it yourself as:
 
 ## Usage
 
+Scraped currently has support for working with HTML and JSON documents.
```
"PRs for other formats are welcome"?

tmtmtmtm commented on README.md:
```diff
@@ -20,14 +22,29 @@ Or install it yourself as:
 
+Scraped currently has support for working with HTML and JSON documents.
+
+To write a scraper, start by creating a subclass of `Scraped::HTML` or `Scraped::JSON` for each _type_ of page you’re scraping. Then specify the (data) fields you want to extract from that. You can control the strategy used to get the page, and decorate the response you get back to make it easier to parse.
```
The emphasis on 'type' makes it seem important, so I think we need something to spell out a bit more what that means (whether inplace here, or a pointer to the section that goes into that)

tmtmtmtm commented on README.md:
```diff
@@ -37,30 +54,43 @@ class ExamplePage < Scraped::HTML
+{
+  name: 'John Doe',
+  email: 'john_doe@example.com'
+}
```
Probably better to use an example that's actually JSON…

Commit:
invite PRs for other formats

considered linking to https://github.com/everypolitician/scraped/pulls
here but I don't think that really adds anything
...but see #96 -- could use a CONTRIBUTING.md on this repo

Commit:
clarify that subclassing the different formats is helpful

It's not so obvious that picking the right one here is about making
it easy to do the next bit (nominate the fields), so add this idea
here.

Commit:
replace JSON example with valid JSON

Can use `json` syntax highlighting too

Commit:
don't imply this is only for multipage scraping

(because it's not)

chrismytton reviewed (approved):
Woo, Thanks @davewhiteland, this is excellent! 💖  LGTM! 🚀 

(And please forgive tardiness of my review! 😥 )
